infra/build/functions/test_data/expected_build_steps.json

```diff
@@ -209,6 +209,143 @@
       "rm -r /workspace/out/afl-address-x86_64"
     ]
   },
+  {
+    "name": "gcr.io/cloud-builders/docker",
+    "env": [
+      "ARCHITECTURE=x86_64",
+      "FUZZING_ENGINE=centipede",
+      "FUZZING_LANGUAGE=c++",
+      "HOME=/root",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "SANITIZER=address"
+    ],
+    "args": [
+      "run",
+      "--platform",
+      "linux/amd64",
+      "-v",
+      "/workspace:/workspace",
+      "-e",
+      "ARCHITECTURE=x86_64",
+      "-e",
+      "FUZZING_ENGINE=centipede",
+      "-e",
+      "FUZZING_LANGUAGE=c++",
+      "-e",
+      "HOME=/root",
+      "-e",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "-e",
+      "SANITIZER=address",
+      "-t",
+      "gcr.io/oss-fuzz/test-project",
+      "bash",
+      "-c",
+      "rm -r /out && cd /src && cd /src && mkdir -p /workspace/out/centipede-address-x86_64 && compile || (echo \"********************************************************************************\nFailed to build.\nTo reproduce, run:\npython infra/helper.py build_image test-project\npython infra/helper.py build_fuzzers --sanitizer address --engine centipede --architecture x86_64 test-project\n********************************************************************************\" && false)"
+    ],
+    "id": "compile-centipede-address-x86_64"
+  },
+  {
+    "name": "gcr.io/cloud-builders/docker",
+    "env": [
+      "ARCHITECTURE=x86_64",
+      "FUZZING_ENGINE=centipede",
+      "FUZZING_LANGUAGE=c++",
+      "HOME=/root",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "SANITIZER=address"
+    ],
+    "args": [
+      "run",
+      "--platform",
+      "linux/amd64",
+      "-v",
+      "/workspace:/workspace",
+      "-e",
+      "ARCHITECTURE=x86_64",
+      "-e",
+      "FUZZING_ENGINE=centipede",
+      "-e",
+      "FUZZING_LANGUAGE=c++",
+      "-e",
+      "HOME=/root",
+      "-e",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "-e",
+      "SANITIZER=address",
```

Comment:
what about the unsanitized build? we need to trigger that automatically too without the user specifying it in project.yaml.

Reply:
Not sure if my understanding is correct:
We don't need another command dedicated to building the unsanitized binary, because when we specify `SANITIZER=address` in this command, the command will also [build the unsanitized binary automatically](https://github.com/google/oss-fuzz/blob/dd6b476ef36e83df1a7461b7ac1793e4fd267c3d/infra/helper.py#L724).

Reply:
I reckon this is similar to the issue above for the same reason: The existing tests are not designed for having an unsanitized binary, and only focus on checking the sanitized one.
Shall I add some other tests for `Centipede` to verify the existence of the unsanitized binaries? Not sure if there is anything else to verify, though.

Reply:
This does not go through the helper.py. The helper.py script goes through a separate path to build the extra binaries.

The build infra here invokes the docker containers to build separately from helper.

Reply:
Done, thanks : )